app/screens/home/search/search.test.tsx

```diff
@@ -0,0 +1,159 @@
+import {searchPosts, searchFiles} from '@actions/remote/search';
+import {renderWithEverything} from '@test/intl-test-helper';
+import TestHelper from '@test/test_helper';
+
+import SearchScreen from './search';
+
+import type {TeamModel} from '@database/models/server';
+import type {Database} from '@nozbe/watermelondb';
+
+// Some subcomponents require react-native-camera-roll, which is not available in the test environment
+jest.mock('@react-native-camera-roll/camera-roll', () => ({}));
+
+jest.mock('@react-navigation/native', () => ({
+    ...jest.requireActual('@react-navigation/native'),
+    useNavigation: () => ({
+        getState: () => ({
+            index: 0,
+            routes: [{params: {searchTerm: ''}}],
+        }),
+    }),
+    useIsFocused: () => true,
+}));
+
+jest.mock('@actions/local/post', () => ({
+    getPosts: jest.fn().mockResolvedValue([]),
+}));
+
+jest.mock('@actions/local/team', () => ({
+    addSearchToTeamSearchHistory: jest.fn(),
+}));
+
+jest.mock('@actions/remote/search', () => ({
+    searchPosts: jest.fn().mockResolvedValue({order: [], matches: {}}),
+    searchFiles: jest.fn().mockResolvedValue({files: [], channels: []}),
+}));
+
+jest.mock('@mattermost/hardware-keyboard', () => ({
+    useHardwareKeyboardEvents: jest.fn(),
+}));
+
+jest.mock('@screens/navigation', () => ({
+    bottomSheet: jest.fn(),
+}));
+
+describe('SearchScreen', () => {
+    const baseProps = {
+        teamId: 'team1',
+        teams: [
+            {id: 'team1', displayName: 'Team 1'},
+            {id: 'team2', displayName: 'Team 2'},
+        ] as TeamModel[],
+        crossTeamSearchEnabled: true,
+    };
+
+    let database: Database;
+    beforeAll(async () => {
+        const server = await TestHelper.setupServerDatabase();
+        database = server.database;
+    });
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('renders search screen correctly', () => {
+        const {getByTestId} = renderWithEverything(
+            <SearchScreen {...baseProps}/>,
+            {database},
+        );
+        expect(getByTestId('search_messages.screen')).toBeTruthy();
+    });
+
+    it('handles search input changes', () => {
+        const {getByTestId} = renderWithEverything(
+            <SearchScreen {...baseProps}/>,
+            {database},
+        );
+
+        const searchInput = getByTestId('navigation.header.search_bar.search.input');
+        fireEvent.changeText(searchInput, 'test search');
+        expect(searchInput.props.value).toBe('test search');
+    });
+
+    it('performs search when submitting', async () => {
+        const {getByTestId} = renderWithEverything(
+            <SearchScreen {...baseProps}/>,
+            {database},
+        );
+
+        const searchInput = getByTestId('navigation.header.search_bar.search.input');
+        fireEvent.changeText(searchInput, 'test search');
+        fireEvent(searchInput, 'submitEditing');
```

Comment:
This looks odd to me.  I thought it should have been.  But maybe there's something I don't know about.

```
  fireEvent.press();
```

Reply:
There's no buttons to trigger the search, this simulates pressing the "enter" key of the keyboard.